src/components/Post/CommentDialogBox.jsx

```diff
@@ -68,14 +69,15 @@ const CommentDialogBox = ({
                     <ReadMore>{userComment.content.text}</ReadMore>
                   </p>
                 </div>
+
                 <div
                   onClick={() => {
                     setOpenToDeleteComment(!openToDeleteComment);
                     setDeleteCommentID(userComment);
                   }}
                 >
                   {user &&
-                    userComment.content.username === user.displayName && (
+                    userComment?.content?.username === user.uid && ( // TODO: instead of uid, we must use username
```

Comment:
instead of comparing uid to username, we should be comparing username, can you please try to do that? As username is not intended to be same as uid forever.
Hint: implementation would be the same as `imgupload` component to get the username.